analyses/mwi/dryspells_trigger/15_mwi_obs_trigger_monitoring.md

```diff
@@ -0,0 +1,154 @@
+```
+
+The first we will do is setup the 3 monitors, which are going to use the 3 separate types of aggregation. We only need to download the data once for each because they all depend on the same raw data from ARC2. Currently, I've kept them using the 2018 monitoring period as an example because we have not reached the period we want to monitor. To monitor in 2022, all we have to do is set `monitoring_start = "2021-01-07"` and `monitoring_end = date.today()`.
+
+```python
+## Global variables for all monitoring
+
+POLY_PATH = os.path.join(
+    os.getenv('AA_DATA_DIR'),
+    'public',
+    'processed',
+    'mwi',
+    'cod_ab',
+    'mwi_drought_adm2.gpkg'
+)
+
+# Admin3 file for mapping
+gdf_adm3 = gpd.read_file(os.path.join(
+    os.getenv('AA_DATA_DIR'),
+    'public',
+    'raw',
+    'mwi',
+    'cod_ab',
+    'mwi_adm_nso_20181016_shp',
+    'mwi_admbnda_adm3_nso_20181016.shp'
+))
+
+gdf_adm3 = gdf_adm3[gdf_adm3.ADM1_PCODE == "MW3"]
+
+MONITORING_START = "2018-01-07"
+MONITORING_END = "2018-03-07"
+RANGE_X = ("32E", "36E")
+RANGE_Y = ("20S", "5S")
+
+# centroid method
+arc2_centr = DrySpells(
+    country_iso3 = "mwi",
+    monitoring_start = MONITORING_START,
+    monitoring_end = MONITORING_END,
+    range_x = RANGE_X,
+    range_y = RANGE_Y
+)
+
+arc2_centr.download_data()
+
+# touching method
+arc2_touch = DrySpells(
+    country_iso3 = "mwi",
+    monitoring_start = MONITORING_START,
+    monitoring_end = MONITORING_END,
+    range_x = RANGE_X,
+    range_y = RANGE_Y,
+    agg_method = "touching"
+)
+
+# approximate mask
+arc2_approx = DrySpells(
+    country_iso3 = "mwi",
+    monitoring_start = MONITORING_START,
+    monitoring_end = MONITORING_END,
+    range_x = RANGE_X,
+    range_y = RANGE_Y,
+    agg_method = "approximate_mask"
+)
+```
+
+Now with each of these, we can just re-process our data and calculate rolling sums and dry spells.
+
+```python
+arc2_centr.aggregate_data(POLY_PATH, "ADM2_PCODE")
+arc2_centr.calculate_rolling_sum()
+arc2_centr.identify_dry_spells()
+
+arc2_touch.aggregate_data(POLY_PATH, "ADM2_PCODE")
+arc2_touch.calculate_rolling_sum()
+arc2_touch.identify_dry_spells()
+
+arc2_approx.aggregate_data(POLY_PATH, "ADM2_PCODE")
+arc2_approx.calculate_rolling_sum()
+arc2_approx.identify_dry_spells()
+```
+
+How many dry spells have we detected thus far in our monitoring period?
+
+```python
+print(
+    f"Centroid method: {arc2_centr.count_dry_spells()}\n"
+    f"Touching method: {arc2_touch.count_dry_spells()}\n"
+    f"Approx m method: {arc2_approx.count_dry_spells()}"
+)
+```
+
+And are we getting close to triggering? This is useful during monitoring. Let's check the number of areas that have gotten 12 days or more cumulative sum below 2mm rainfall.
+
+```python
+DAYS = 12
+
+print(
+    f"Centroid method: {arc2_centr.count_days_under_threshold(DAYS)}\n"
+    f"Touching method: {arc2_touch.count_days_under_threshold(DAYS)}\n"
+    f"Approx m method: {arc2_approx.count_days_under_threshold(DAYS)}"
+)
+```
```

Comment:
not super important but might be nice to add the max number of cons days observed in an area? 

`print(
    f"Centroid method: {arc2_centr.days_under_threshold(raster=False).days_under_threshold.max()}\n"
    f"Touching method: {arc2_touch.days_under_threshold(raster=False).days_under_threshold.max()}\n"
    f"Approx m method: {arc2_approx.days_under_threshold(raster=False).days_under_threshold.max()}"
)`